manim/mobject/svg/svg_mobject.py

```diff
@@ -128,116 +140,133 @@ def generate_points(self):
         the SVGMobject's points from XML tags, populating self.mobjects, and
         any submobjects within self.mobjects.
         """
-        doc = minidom.parse(self.file_path)
-        self.ref_to_element = {}
+        doc = minidom_parse(self.file_path)
         for svg in doc.getElementsByTagName("svg"):
-            mobjects = self.get_mobjects_from(svg)
+            mobjects = self.get_mobjects_from(svg, self.DEFAULT_SVG_STYLE)
             if self.unpack_groups:
                 self.add(*mobjects)
             else:
                 self.add(*mobjects[0].submobjects)
         doc.unlink()
 
-    def get_mobjects_from(self, element):
+    def get_mobjects_from(
+        self,
+        element: MinidomElement,
+        inherited_style: Dict[str, str],
+        within_defs: bool = False,
+    ) -> List[VMobject]:
         """Parses a given SVG element into a Mobject.
 
         Parameters
         ----------
-        element : :class:`str`
+        element : :class:`Element`
             The SVG data in the XML to be parsed.
 
+        inherited_style : :class:`dict`
+            Dictionary of the SVG attributes for children to inherit.
+
+        within_defs : :class:`bool`
+            Whether ``element`` is within a ``defs`` element, which indicates
+            whether elements with `id` attributes should be added to the
+            definitions list.
+
         Returns
         -------
-        VMobject
+        List[VMobject]
             A VMobject representing the associated SVG element.
         """
+
         result = []
-        if not isinstance(element, minidom.Element):
+        # First, let all non-elements pass (like text entries)
+        if not isinstance(element, MinidomElement):
             return result
-        if element.tagName == "defs":
-            self.update_ref_to_element(element)
-        elif element.tagName == "style":
+
+        style = self.cascade_element_style(element, inherited_style)
+        is_defs = element.tagName == "defs"
+
+        if element.tagName == "style":
             pass  # TODO, handle style
-        elif element.tagName in ["g", "svg", "symbol"]:
+        elif element.tagName in ["g", "svg", "symbol", "defs"]:
```

Comment:
The idea is to parse the elements in defs just as normal, but instead of adding them to the main group at the end, just go through step-by-step and add anything with an ID to the defs_to_mobjects map.

And yeah, you're spot-on - the issue is with gradients, not with defs, fortunately.

I don't like that there aren't any defs in the SVG tests, but MarkupText tests certainly use them, so that's some consolation.